src/Stratis.Bitcoin/Consensus/ConsensusRuleEngine.cs

```diff
@@ -226,17 +232,26 @@ private async Task ExecuteRulesAsync(IEnumerable<AsyncConsensusRule> asyncRules,
             catch (ConsensusErrorException ex)
             {
                 ruleContext.ValidationContext.Error = ex.ConsensusError;
+            }
+        }
 
-                if (ruleContext.ValidationContext.Error != ConsensusErrors.BadTransactionDuplicate)
-                {
-                    uint256 hashToBan = ruleContext.ValidationContext.ChainedHeaderToValidate.HashBlock;
+        /// <summary>Adds block hash to a list of failed header unless specific consensus error was used that doesn't require block banning.</summary>
+        private void HandleConsensusError(ValidationContext validationContext)
+        {
+            this.logger.LogTrace("()");
 
-                    if (ruleContext.ValidationContext.RejectUntil != null)
-                        this.invalidBlockHashStore.MarkInvalid(hashToBan, ruleContext.ValidationContext.RejectUntil);
-                    else
-                        this.invalidBlockHashStore.MarkInvalid(hashToBan);
-                }
+            if (validationContext.Error == ConsensusErrors.BadWitnessNonceSize)
```

Comment:
Probably a comment would be nice saying that this is handled in consensus manager and why the block should not be banned

Reply:
fixed